src/pid/environ.rs

```diff
@@ -0,0 +1,179 @@
+//! Process initial environment from `/proc/[pid]/environ`.
+
+use std::ffi::{OsStr, OsString};
+use std::fs::File;
+use std::io::{Result, Read, Error, ErrorKind};
+use std::os::unix::ffi::OsStrExt;
+use std::path::Path;
+use std::iter::Iterator;
+
+use libc::pid_t;
+use nom::{self, IResult};
+
+/// An environment holder.
+#[derive(Debug, Clone)]
+pub struct Environ {
+    data: Vec<u8>,
+}
+
+/// An iterator over environment variables.
+pub struct EnvironIter<'a> {
```

Comment:
Can `Environ` and `EnvironIter` be combined into a single type?  I think it should be possible just using a `Vec<u8>` data field and a `usize` offset.